Tests/TestingTests/TestSupport/TestingAdditions.swift

```diff
@@ -290,6 +290,29 @@ extension Test.ID.Selection {
   }
 }
 
+extension Configuration {
+  /// Set an event handler which automatically handles thrown errors.
+  ///
+  /// - Parameters:
+  ///   - eventHandler: The throwing ``Event/Handler`` to set.
+  ///
+  /// Errors thrown by `eventHandler` are caught and recorded as an ``Issue``.
+  ///
+  /// This is meant for testing the testing library itself. In production tests,
+  /// event handlers should not typically need to be throwing, but if they do,
+  /// the event handler should implement its own error-handling logic since
+  /// recording an ``Issue`` would be inappropriate.
+  mutating func setEventHandler(_ eventHandler: @escaping @Sendable (_ event: borrowing Event, _ context: borrowing Event.Context) throws -> Void) {
```

Comment:
Was this change intentional? Should the error-handling logic be part of the overall interface instead?

Reply:
I tried to explain my rationale for why I don't think it should be built-in in the last paragraph of the DocC comment above. In particular, it doesn't seem appropriate for a "production" event handler to record `Issue`s itself, so I think it's reasonable to say that event handler closures must catch & handle errors on their own. It's just that we're a special case here because we're using a nested event handler so we _can_ safely record `Issue`s from our "inner" event handler.

Reply:
🤔 I see what you mean. I do wish we had the ability to make the SPI event handler throwing as well, but as you say the outermost runner isn't going to have anywhere to record issues to.

Not for this PR, but… we _could_ set up the default state to `fatalError()` while allowing nested handlers to propagate errors outward, if we wanted to.